src/database_impl/setup.rs

```diff
@@ -0,0 +1,147 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+use std::{
+    borrow::{Borrow, Cow},
+    fmt::Display,
+    sync::{Arc, Mutex},
+};
+
+use async_trait::async_trait;
+use mysql::{prelude::Queryable, Conn, OptsBuilder, Row};
+
+use crate::{config::DatabaseConnConfig, error::Result, Database, SqlnessError};
+
+#[async_trait]
+pub trait DatabaseBuilder: Send + Sync + Default {
```

Comment:
Why this trait depend on Default?

Reply:
As with `build_default_database` above, I need Default to be compatible with other database implementations

Reply:
I still can't understand why this is required.

Mysql builder should have no relation with other DB.

Reply:
That's what I think. If additional database support is added later, the Builder can be implemented like this:
```
pub struct PostgreSQLBuilder;

#[async_trait]
impl DatabaseBuilder for PostgreSQLBuilder {
    async fn build(&self, config: DatabaseConnConfig) -> Result<Box<dyn Database>> {
        todo!()
    }
}
```
We can then create a default implementation of PostgreSQL like this:
```
let pg_db = build_default_database::<PostgreSQLBuilder>()
```
Maybe I should delete them now😂